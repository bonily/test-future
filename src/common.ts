import {PersonType} from './types';
import {Values} from './blocks/person-form/person-form';
import {SORT_TYPES} from "./const";

export const extend = (a : any, b: any): any => {
  return Object.assign({}, a, b);
};

export const getPersonsBySortType = (persons : PersonType[], sortType : string, ascending : boolean): PersonType[] => {
  switch (sortType) {
    case SORT_TYPES.FIRST_NAME:
      return persons.sort((a, b) => ascending ? a.firstName.localeCompare(b.firstName) : b.firstName.localeCompare(a.firstName)).slice();
    case SORT_TYPES.LAST_NAME:
      return persons.sort((a, b) => ascending ? a.lastName.localeCompare(b.lastName) : b.lastName.localeCompare(a.lastName)).slice();
    case SORT_TYPES.EMAIL:
      return persons.sort((a, b) => ascending ? a.email.localeCompare(b.email) : b.email.localeCompare(a.email)).slice();
    case SORT_TYPES.ID:
      return persons.sort((a, b) => ascending ? a.id - b.id : b.id - a.id).slice();
    case SORT_TYPES.PHONE:
      return persons.sort((a, b) => ascending ? a.phone.localeCompare(b.phone) : b.phone.localeCompare(a.phone)).slice();
  }
  return persons.slice();
};

export const noop = () : void => {
  // do nothing
};

export const getFilteredPersons = (persons : PersonType[], str = ``):PersonType[] => {
  return persons.filter((person) =>
    person.firstName.toLowerCase().indexOf(str) > -1 ||
    String(person.id).indexOf(str) > -1 ||
    person.lastName.toLowerCase().indexOf(str) > -1 ||
    person.email.toLowerCase().indexOf(str) > -1 ||
    person.phone.toLowerCase().indexOf(str) > -1 ||
    person.address.streetAddress.toLowerCase().indexOf(str) > -1 ||
    person.address.state.toLowerCase().indexOf(str) > -1 ||
    person.address.zip.toLowerCase().indexOf(str) > -1 ||
    person.address.city.toLowerCase().indexOf(str) > -1);
};

// export const updateFavoriteStatus = (persons : PersonType[], id : number):PersonType[] => {
//   const index = persons.findIndex((person) => person.id === id);
//   persons[index].favourite = !persons[index].favourite;
//   return persons;
// };

export const castFormValueToPersonType = (value: Values) : PersonType => {
  return {
    id: Number(value.id),
    firstName: value.firstName,
    lastName: value.lastName,
    email: value.email,
    phone: value.phone,
    address: {
      streetAddress: value.streetAddress,
      city: value.city,
      state: value.state,
      zip: value.zip
    },
    description: value.description,
  };
};
