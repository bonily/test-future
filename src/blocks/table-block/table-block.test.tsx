import React from 'react';
import * as renderer from "react-test-renderer";
import {noop} from "../../common";
import TableBlock from "./table-block";

const persons = [
  {
    address: {
      city: `Bradford`,
      state: `TX`,
      streetAddress: `7426 Suspendisse Dr`,
      zip: `14783`,
    },
    description: `pulvinar magna convallis quis porta adipiscing mi porta vestibulum elit amet ipsum egestas tortor suspendisse mattis rutrum lacus vel libero vitae sit dolor molestie lorem sed massa lacus lacus vitae placerat et`,
    email: `MRamsey@scelerisque.com`,
    firstName: `Jeri`,
    id: 714,
    lastName: `Wittcop`,
    phone: `(731)368-3302`
  },
  {
    address: {
      city: `Bradford`,
      state: `TX`,
      streetAddress: `7426 Suspendisse Dr`,
      zip: `14745`,
    },
    description: `pulvinar magna convallis quis porta adipiscing mi porta vestibulum elit amet ipsum egestas tortor suspendisse mattis rutrum lacus vel libero vitae sit dolor molestie lorem sed massa lacus lacus vitae placerat et`,
    email: `MRamsey@scelerisque.com`,
    firstName: `Bonya`,
    id: 120,
    lastName: `Wittcop`,
    phone: `(731)368-3302`
  }
];

const sortType = `email`;

it(`Person form component should be render with person array length > 0`, () => {
  const tree = renderer
  .create(<TableBlock
    persons = {persons}
    sortType = {sortType}
    onPersonTrClick = {noop}
    onSortTitleClick = {noop}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});


it(`Person form component shoul be render with person array length < 0`, () => {
  const tree = renderer
  .create(<TableBlock
    persons = {[]}
    sortType = {sortType}
    onPersonTrClick = {noop}
    onSortTitleClick = {noop}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
