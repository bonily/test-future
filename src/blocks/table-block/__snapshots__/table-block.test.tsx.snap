// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person form component shoul be render with person array length < 0 1`] = `
<table
  className="sc-dlfnbm kIMfcW"
>
  <colgroup>
    <col
      style={
        Object {
          "background": "yellow",
        }
      }
    />
    <col />
    <col />
    <col />
    <col />
  </colgroup>
  <tbody>
    <tr
      className="sc-hKgILt"
    >
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        id
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon.svg"
          width="18px"
        />
      </th>
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        firstName
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon.svg"
          width="18px"
        />
      </th>
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        lastName
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon.svg"
          width="18px"
        />
      </th>
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        email
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon--active.svg"
          width="18px"
        />
      </th>
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        phone
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon.svg"
          width="18px"
        />
      </th>
    </tr>
    <tr
      className="sc-hKgILt"
    >
      <td
        className="sc-eCssSg fGPNoo"
        colSpan={5}
        style={
          Object {
            "background": "pink",
          }
        }
      >
        Ничего не найдено
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Person form component should be render with person array length > 0 1`] = `
<table
  className="sc-dlfnbm kIMfcW"
>
  <colgroup>
    <col
      style={
        Object {
          "background": "yellow",
        }
      }
    />
    <col />
    <col />
    <col />
    <col />
  </colgroup>
  <tbody>
    <tr
      className="sc-hKgILt"
    >
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        id
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon.svg"
          width="18px"
        />
      </th>
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        firstName
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon.svg"
          width="18px"
        />
      </th>
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        lastName
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon.svg"
          width="18px"
        />
      </th>
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        email
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon--active.svg"
          width="18px"
        />
      </th>
      <th
        className="sc-bdfBwQ khoPZA"
        onClick={[Function]}
      >
        phone
        <img
          className="sc-gsTCUz bqElTy"
          height="18px"
          src="./img/sort-icon.svg"
          width="18px"
        />
      </th>
    </tr>
    <tr
      className="sc-hKgILt"
      onClick={[Function]}
    >
      <td
        className="sc-eCssSg fGPNoo"
      >
        714
      </td>
      <td
        className="sc-eCssSg fGPNoo"
      >
        Jeri
      </td>
      <td
        className="sc-eCssSg fGPNoo"
      >
        Wittcop
      </td>
      <td
        className="sc-eCssSg fGPNoo"
      >
        MRamsey@scelerisque.com
      </td>
      <td
        className="sc-eCssSg fGPNoo"
      >
        (731)368-3302
      </td>
    </tr>
    <tr
      className="sc-hKgILt"
      onClick={[Function]}
    >
      <td
        className="sc-eCssSg fGPNoo"
      >
        120
      </td>
      <td
        className="sc-eCssSg fGPNoo"
      >
        Bonya
      </td>
      <td
        className="sc-eCssSg fGPNoo"
      >
        Wittcop
      </td>
      <td
        className="sc-eCssSg fGPNoo"
      >
        MRamsey@scelerisque.com
      </td>
      <td
        className="sc-eCssSg fGPNoo"
      >
        (731)368-3302
      </td>
    </tr>
  </tbody>
</table>
`;
